web/packages/design/src/CollapsibleInfoSection/CollapsibleInfoSection.tsx

```diff
@@ -0,0 +1,140 @@
+  useState,
+} from 'react';
+import styled from 'styled-components';
+
+import { Box, Button, Text } from 'design';
+import { BoxProps } from 'design/Box';
+import { Minus, Plus } from 'design/Icon';
+
+type CollapsibleInfoSectionProps = {
+  /* defaultOpen is optional and determines whether the section is open or closed initially */
+  defaultOpen?: boolean;
+  /* onClick is an optional callback for when the toggle is clicked */
+  onClick?: (isOpen: boolean) => void;
+  /* openLabel is an optional label for the closed state */
+  openLabel?: string;
+  /* closeLabel is an optional label for the opened state */
+  closeLabel?: string;
+  /* disabled is an optional flag to disable the toggle */
+  disabled?: boolean;
+} & BoxProps;
+
+/**
+ * CollapsibleInfoSection is a collapsible section that shows more information when expanded.
+ * It is useful for hiding less important – or more detailed – information by default.
+ */
+export const CollapsibleInfoSection = ({
+  defaultOpen = false,
+  openLabel = 'More info',
+  closeLabel = 'Less info',
+  onClick,
+  disabled = false,
+  children,
+  ...boxProps
+}: PropsWithChildren<CollapsibleInfoSectionProps>) => {
+  const contentId = useId();
+  const [isOpen, setIsOpen] = useState(defaultOpen);
+  const [contentHeight, setContentHeight] = useState(0);
+  const [shouldRenderContent, setShouldRenderContent] = useState(defaultOpen);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    if (!contentRef.current || !shouldRenderContent) {
+      return;
+    }
+    const ro = new ResizeObserver(entries => {
+      for (const entry of entries) {
+        if (entry.target === contentRef.current) {
+          setContentHeight(entry.contentRect.height);
+        }
+      }
+    });
+    ro.observe(contentRef.current);
+    return () => ro.disconnect();
+  }, [contentRef, shouldRenderContent]);
+
+  return (
+    <Box {...boxProps}>
+      <ToggleButton
+        onClick={() => {
+          !isOpen && setShouldRenderContent(true);
+          setIsOpen(!isOpen);
+          onClick?.(!isOpen);
+        }}
+        disabled={disabled}
+        aria-expanded={isOpen}
+        aria-controls={contentId}
+      >
+        {isOpen ? <Minus size="small" /> : <Plus size="small" />}
+        <Text>{isOpen ? closeLabel : openLabel}</Text>
+      </ToggleButton>
+      <ContentWrapper
+        id={contentId}
+        role="region"
+        aria-hidden={!isOpen}
+        $isOpen={isOpen}
+        $contentHeight={contentHeight}
+        onTransitionEnd={() => !isOpen && setShouldRenderContent(false)}
+      >
+        {shouldRenderContent && (
+          <Box ml={4} style={{ position: 'relative' }} ref={contentRef}>
+            <Bar />
+            <Box ml={3} pt={3} pb={2}>
+              {children}
+            </Box>
+          </Box>
+        )}
+      </ContentWrapper>
+    </Box>
+  );
+};
+
+const ToggleButton = styled(Button).attrs({
+  intent: 'neutral',
+})`
+  display: flex;
```

Comment:
nit: that indention makes it a bit off place. 
```suggestion
const ToggleButton = styled(Button).attrs({intent: 'neutral'})`
  display: flex;
```

(unless that is applied by prettier)

Reply:
That was probably Prettier's doing, can try making it one line though.